app/models/download.rb

```diff
@@ -212,12 +212,8 @@ def self.rank(version)
   end
 
   def self.highest_rank(versions)
-    ranks = versions.map { |version| Download.rank(version) }.reject(&:zero?)
-    if ranks.empty?
-      0
-    else
-      ranks.min
-    end
+    ranks = versions.map { |version| rank(version) }.reject(&:zero?)
+    ranks.empty? ? 0 : ranks.min
```

Comment:
What a weird section of code. We reject all entries that are zero - but then if we don't find any, zero is the default. Can we just use `min { |version| rank(version)` ?  Also `#min` returns nil if there's nothing in the array, so we don't need the ternary at all.